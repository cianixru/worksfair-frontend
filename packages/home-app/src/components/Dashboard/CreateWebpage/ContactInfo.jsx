import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import ContactInfoForm from '../../../forms/Webpage/ContactInfoForm';

const ContactInfo = ({
  onSubmit, validationErrors, handleErrorReset, webpage,
}) => {
  return (
    <Fragment>
      <div className="add-webpage-form">
        <ContactInfoForm
          onSubmit={onSubmit}
          validationErrors={validationErrors}
          handleErrorReset={handleErrorReset}
          webpage={webpage} />
      </div>
    </Fragment>
  );
};

ContactInfo.propTypes = {
  onSubmit: PropTypes.func,
  user: PropTypes.object,
  validationErrors: PropTypes.object,
  handleErrorReset: PropTypes.func,
  webpage: PropTypes.object,
};

export default ContactInfo;
