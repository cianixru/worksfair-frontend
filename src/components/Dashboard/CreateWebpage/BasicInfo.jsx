import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import CreateWebpageForm from '../../../forms/Webpage/CreateWebpageForm';
import avatar from '../../../assets/worksfair-avatar.png';
import { toSentenceCase } from '../../utils/helpers';

const BasicInfo = ({
  onSubmit, user, validationErrors, handleErrorReset, webpage,
}) => {
  return (
    <Fragment>
      <h5>Website Owner</h5>
      <div className="media">
        <div className="media-left">
          <figure className="image is-32x32">
            <img
              src={user && user.image_url ? user.image_url : avatar}
              alt="avatar"
              className="is-rounded"
            />
          </figure>
        </div>
        <div className="media-content">
          { user
              && `${toSentenceCase(user.first_name)} ${toSentenceCase(user.last_name)}`
          }
        </div>
      </div>
      {/* <div className="margin-top-20">
        <h5>Webpage Subdomain name</h5>
        <p>{webpage && webpage.sub_domain_name}</p>
      </div> */}
      <div className="add-webpage-form">
        <CreateWebpageForm
          onSubmit={onSubmit}
          validationErrors={validationErrors}
          handleErrorReset={handleErrorReset}
          webpage={webpage}
        />
      </div>
    </Fragment>
  );
};

BasicInfo.propTypes = {
  onSubmit: PropTypes.func,
  user: PropTypes.object,
  validationErrors: PropTypes.object,
  handleErrorReset: PropTypes.func,
  webpage: PropTypes.object,
};

export default BasicInfo;
