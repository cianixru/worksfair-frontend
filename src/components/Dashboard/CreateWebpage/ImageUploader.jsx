import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import axios from 'axios';
import PropTypes from 'prop-types';


import { CloudinaryImageUploader } from '../../utils/helpers';

class ImageUploader extends Component {
  state={
    selectedImages: [],
    uploadedImages: [],
    rawFiles: {},
  }

  /**
   * @description Handles the selection of the images
   *
   * @param { Array } images
   *
   * @returns { Promise } axios API call
   */

  handleImagesSelection = (files) => {
    const rawFiles = {};
    const localFileURLs = files.map((file) => {
      const localFileURL = URL.createObjectURL(file);
      rawFiles[localFileURL] = file;
      return localFileURL;
    });

    // console.log(files, localFileURLs, rawFiles);
    const { selectedImages } = this.state;
    this.setState({
      ...selectedImages,
      selectedImages: localFileURLs,
      rawFiles,
    });
  }

  /**
   * @description Handles the response from cloudinary API call
   *
   * @param { array } images
   */
  handleCloudinaryResponse = async (images) => {
    const { uploadedImages } = this.state;
    const uploads = images.map((image) => {
      return CloudinaryImageUploader(image)
        .then((imageData) => {
          uploadedImages.push(imageData.data.secure_url);
          this.setState({
            uploadedImages,
          });
        });
    });
    return uploads;
  }

  /**
   * @description Handles the removal of uploaded image from preview
   *
   * @param { string } imageURL
   *
   * @returns { Promise } axios API call
   */
  handleRemoveImage = imageURL => () => {
    const { selectedImages } = this.state;

    // find and remove the image from the selectedImages array
    selectedImages.splice(selectedImages.indexOf(imageURL), 1);

    this.setState({
      selectedImages,
    });
  }

  /**
   * @description sends the cloudinary urls to the DB
   */
  handleSubmit = async () => {
    const { uploadedImages, selectedImages, rawFiles } = this.state;

    const imagesToUpload = [];
    selectedImages.map((imageBlob) => {
      return imagesToUpload.push(rawFiles[imageBlob]);
    });

    const imagesData = await this.handleCloudinaryResponse(imagesToUpload);
    axios.all(imagesData).then(() => {
      const images = {
        featured_images: uploadedImages,
      };
      this.props.onSubmit(images);
      console.log('Images have all being uploaded to cloudinary');
    });
  }

  render() {
    const { selectedImages } = this.state;
    const hasNoImages = selectedImages.length <= 0;

    return (
      <Dropzone
        onDrop={this.handleImagesSelection}
        multiple
        accept="image/*"
      >
        {({ getRootProps, getInputProps }) => {
          const pictureList = selectedImages.map(localObjectURL => (
            <li
              key={localObjectURL}
              className="notification column is-one-quarter"
            >
              <button
                className="delete"
                onClick={this.handleRemoveImage(localObjectURL)}
                data-testid="delete-image"
              />
              <figure className="image is-180x180">
                <img
                  src={localObjectURL}
                  alt={localObjectURL}
                />
              </figure>
            </li>
          ));

          return (
            <div>
              <div className="drop-area">
                <div
                  {...getRootProps({ className: 'dropzone' })}
                >
                  <input {...getInputProps()} />
                  <p>Try dropping some files here, or click to select files to upload.</p>
                </div>
                <ul className="columns is-multiline ">{pictureList}</ul>
              </div>
              <div className="control">
                <button
                  type="submit"
                  disabled={hasNoImages}
                  className="button is-danger is-medium is-rounded"
                  data-testid="save-featured-images"
                  onClick={this.handleSubmit}
                >
                  Save & Continue
                </button>
              </div>
            </div>
          );
        }}
      </Dropzone>
    );
  }
}

ImageUploader.propTypes = {
  onSubmit: PropTypes.func,
};

export default ImageUploader;
