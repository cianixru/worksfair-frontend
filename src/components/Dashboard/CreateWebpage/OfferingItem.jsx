import React from 'react';
import PropTypes from 'prop-types';

import { addNaira } from '../../utils/helpers';

const OfferingItem = ({
  offerings,
}) => {
  return (
    <ul>
      { offerings.length > 0
        && offerings.map(offering => (
          <li
            className="media box"
            key={offering.title}
          >
            <div className="media-left">
              <figure className="image is-180x180">
                <img
                  src={offering.image}
                  alt=""
                  className="offering-image"
                />
              </figure>
            </div>
            <div className="media-content">
              <h4 className="title is-5">
                {offering.title}
              </h4>
              <p>{offering.description}</p>
              <div className="">
                {addNaira(offering.price)}
              </div>
            </div>
          </li>
        ))
      }
    </ul>
  );
};

OfferingItem.propTypes = {
  offerings: PropTypes.array,
};

export default OfferingItem;
