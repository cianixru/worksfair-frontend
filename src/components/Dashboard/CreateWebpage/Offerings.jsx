import React from 'react';
import PropTypes from 'prop-types';

import ImageUpload from '../../utils/ImageUpload';
import OfferingsForm from '../../../forms/Webpage/OfferingsForm';
import OfferingItems from './OfferingItem';

const Offerings = ({
  onSubmit,
  validationErrors,
  handleErrorReset,
  handleOfferingImageSelection,
  offerings,
  offeringsFormRef,
  selectedImage,
  handleSaveAndPreview,
  webpage,
  onUpdateOffering,
  handleDelete,
}) => {
  return (
    <div>
      <div className="margin-bottom-25">
        <OfferingItems
          offerings={(webpage && webpage.offerings) || offerings}
          validationErrors={validationErrors}
          handleErrorReset={handleErrorReset}
          handleOfferingImageSelection={handleOfferingImageSelection}
          selectedImage={selectedImage}
          onSubmit={onUpdateOffering}
          colour={webpage && webpage.colour}
          handleDelete={handleDelete}
        />
      </div>
      <div className="columns box margin-bottom-25">
        <div className="column is-one-third">
          <ImageUpload
            handleOfferingImageSelection={handleOfferingImageSelection}
            selectedImage={selectedImage}
          />
        </div>
        <div className="column">
          <OfferingsForm
            onSubmit={onSubmit}
            validationErrors={validationErrors}
            offeringsFormRef={offeringsFormRef}
            handleErrorReset={handleErrorReset}
          />
        </div>
      </div>
      <div className="control">
        <button
          type="submit"
          // disabled={pristine || invalid}
          className="button is-medium is-danger is-rounded"
          data-testid="save-and-preview"
          onClick={handleSaveAndPreview}
        >
          Preview Webpage
        </button>
      </div>
    </div>
  );
};

Offerings.propTypes = {
  onSubmit: PropTypes.func,
  onUpdateOffering: PropTypes.func,
  user: PropTypes.object,
  validationErrors: PropTypes.object,
  handleErrorReset: PropTypes.func,
  handleOfferingImageSelection: PropTypes.func,
  offerings: PropTypes.array,
  offeringsFormRef: PropTypes.object,
  selectedImage: PropTypes.string,
  handleSaveAndPreview: PropTypes.func,
  handleDelete: PropTypes.func,
  webpage: PropTypes.object,
};

export default Offerings;
