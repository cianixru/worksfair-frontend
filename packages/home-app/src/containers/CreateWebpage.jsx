import React, { Component } from 'react';
import { Helmet } from 'react-helmet';

import CreateWebpage from '../components/Dashboard/CreateWebpage';
import DashboardHeader from '../components/Dashboard/DashboardHeader';
import Footer from '../components/Footer/Footer';

class CreateWebpageContainer extends Component {
  render() {
    const username = localStorage.getItem('username');
    const navigation = [
      {
        text: 'Dashboard',
        to: `/dashboard/${username}/webpages`,
      },
      {
        text: 'Create Webpage',
        to: `/dashboard/${username}/webpages/new/basic-info`,
      },
    ];
    return (
      <div className="dashboard-content">
        <Helmet>
          <title>Create Webpage - Worksfair</title>
        </Helmet>
        <DashboardHeader
          title="Create Webpage"
          navigation={navigation}
        >
          <a
            href="http://ideosynergy.worksfair.com"
            target="_blank"
            rel="noopener noreferrer"
          >
            View a sample business webpage
          </a>
        </DashboardHeader>
        <CreateWebpage username={username} />
        <Footer/>
      </div>
    );
  }
}

export default CreateWebpageContainer;
