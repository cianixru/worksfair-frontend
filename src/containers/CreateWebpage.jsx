import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import CreateWebpage from '../components/Dashboard/CreateWebpage';
import DashboardHeader from '../components/Dashboard/DashboardHeader';

class CreateWebpageContainer extends Component {
  render() {
    const username = localStorage.getItem('username');
    const navigation = [
      {
        text: 'Dashboard',
        to: `/dashboard/${username}`,
      },
      {
        text: 'Create Webpage',
        to: `/dashboard/${username}/webpages/new`,
      },
    ];
    return (
      <div className="dashboard-content">
        <DashboardHeader
          title="Create Webpage"
          navigation={navigation}
        >
          <Link to="/demo">View Sample Webpage</Link>
        </DashboardHeader>
        <CreateWebpage />
      </div>
    );
  }
}

export default CreateWebpageContainer;
