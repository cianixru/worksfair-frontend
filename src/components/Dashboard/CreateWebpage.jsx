/* eslint-disable react/jsx-no-bind */
import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  withRouter, NavLink, BrowserRouter as Router, Route
} from 'react-router-dom';

import BasicInfo from './CreateWebpage/BasicInfo';
import ContactInfo from './CreateWebpage/ContactInfo';
import {
  createWebpage,
  updateWebpage,
  createWebpageOffering,
  CREATE_WEBPAGE_FAILED,
  UPDATE_WEBPAGE_FAILED,
  CREATE_OFFERING_FAILED,
} from '../../actions/webpage';
import alert from '../utils/alert';
import ImageUploader from './CreateWebpage/ImageUploader';
import Offerings from './CreateWebpage/Offerings';
import { cloudinaryImageUploader } from '../utils/helpers';
import { isLoading, isComplete } from '../../actions/loader';

class CreateWebpage extends Component {
  state = {
    validationErrors: {
      title: [],
      description: [],
      keywords: [],
    },
    imageArray: null,
    uploadedImage: null,
    selectedOfferingImage: null,
    offerings: [],
  }

  offeringsFormRef = React.createRef();

  /**
   * @description handles submit action when the save button is clicked
   * @param { object } input
   */
  onSubmit = async (input) => {
    const { actions, history, user } = this.props;
    try {
      const response = await actions.createWebpage(input);
      if (response.type === CREATE_WEBPAGE_FAILED) {
        const { data } = response.response;
        this.setState({
          validationErrors: data,
        });
        alert.error('Request Failed. Check for more details');
      } else {
        alert.success('Successful. Keep going!');
        history.push(`/dashboard/${user.username}/webpages/new/contact-info`);
      }
    } catch (error) {
      alert.error(error.message);
    }
  }

  /**
   * @description handles the reset of the error detail
   * @param { string } name
   */
  handleErrorReset = (name) => {
    this.setState({
      validationErrors: {
        ...this.state.validationErrors,
        [name]: [],
      },
    });
  };

  /**
   * @description handles submit action for webpage updates
   *
   * @param { object } input
   */
  submitContactInfo = async (input) => {
    const {
      actions, history, webpage, user
    } = this.props;
    try {
      if (!webpage.sub_domain_name) {
        alert.error(
          'An error occured. Don\'t worry you can continue from where you stopped'
        );
        setTimeout(() => {
          window.location.pathname = `/dashboard/${user.username}/webpages`;
        }, 4000);
      }
      input.subDomainName = webpage.sub_domain_name;
      const response = await actions.updateWebpage(input);
      if (response.type === UPDATE_WEBPAGE_FAILED) {
        const { data } = response.response;
        this.setState({
          validationErrors: data,
        });
        alert.error('Request Failed. Check for more details');
      } else {
        alert.success('Successful. Keep going!');
        history.push(`/dashboard/${user.username}/webpages/new/gallery`);
      }
    } catch (error) {
      alert.error(error.message);
    }
  };

  /**
   * @description Handles the submission of the images
   *
   * @param { object } images
   */
  submitImages = async (images) => {
    const {
      actions, history, webpage, user
    } = this.props;
    try {
      images.subDomainName = webpage.sub_domain_name;
      const response = await actions.updateWebpage(images);
      if (response.type === UPDATE_WEBPAGE_FAILED) {
        const { data } = response.response;
        this.setState({
          validationErrors: data,
        });
        alert.error('Request Failed. Check for more details');
      } else {
        alert.success('Successful. Keep going!');
        history.push(`/dashboard/${user.username}/webpages/new/offerings`);
      }
    } catch (error) {
      alert.error(error.message);
    }
  };

  /**
   * @description handles the selection of an image
   *
   * @param { array } imageArray
   */
  handleOfferingImageSelection = (imageArray) => {
    // convert the image file to object URL for preview
    const localFileURL = URL.createObjectURL(imageArray[0]);
    this.setState({
      selectedImage: localFileURL,
      imageArray,
    });
  }

  /**
   * @description Handles the reset of image on offering form
   *
   * @param { Array } images
   */
  resetOfferingImage = () => {
    this.setState({
      selectedImage: null,
      imageArray: null,
    });
  }

  /**
   * @description Handles the submission of the offerings
   *
   * @param { object } input
   */
  submitOfferings = async (input) => {
    const { imageArray, offerings } = this.state;
    const {
      actions, webpage, user
    } = this.props;

    if (!webpage.sub_domain_name) {
      alert.error(
        'An error occured. Don\'t worry you can continue from where you stopped'
      );
      setTimeout(() => {
        window.location.pathname = `/dashboard/${user.username}/webpages`;
      }, 4000);
    }
    actions.isLoading();
    if (!imageArray) {
      alert.error('You need to upload an image');
    }
    try {
      input.subDomainName = webpage.sub_domain_name;

      // Upload the image to cloudinary and get the response data
      const imageData = await cloudinaryImageUploader(imageArray[0]);
      input.image = imageData.data.secure_url;

      // Send the input and the image URL from cloudinary to the DB
      const response = await actions.createWebpageOffering(input);

      if (response.type === CREATE_OFFERING_FAILED) {
        const { data } = response.response;
        this.setState({
          validationErrors: data,
        });
        alert.error('Request Failed. Check for more details');
      } else {
        offerings.push(response.data);
        alert.success('Successful. Go ahead and preview your webpage!');
        this.setState({
          offerings,
        });

        // Reset the form and the image preview for fresh input
        this.offeringsFormRef.current.reset();
        this.resetOfferingImage();
      }
    } catch (error) {
      console.log(error);
    } finally {
      actions.isComplete();
    }
  };

  /**
   * @description Handles the submission of the offerings
   *
   * @param { object } input
   */
  handleSaveAndPreview = async () => {
    const { webpage, actions, user } = this.props;

    if (!webpage.sub_domain_name) {
      alert.error(
        'An error occured. Don\'t worry you can continue from where you stopped'
      );
      setTimeout(() => {
        window.location.pathname = `/dashboard/${user.username}/webpages`;
      }, 4000);
    }

    actions.isLoading();
    // Make app wait for 3 seconds to simulate running background process
    // good for user experience
    setTimeout(() => {
      actions.isComplete();
      window.location.pathname = `webpage/${webpage.sub_domain_name}`;
    }, 3000);
  }

  /**
   * @description disables the Navlink conditionally
   *
   * @param { object } event DOM event
   */
  handleNavlinkClick = (event) => {
    const { webpage } = this.props;

    if (!webpage.sub_domain_name) {
      event.preventDefault();
    }
  }

  render() {
    const { validationErrors, offerings, selectedImage } = this.state;

    const {
      user, username, webpage, actions,
    } = this.props;
    const others = {
      isComplete: actions.isComplete,
      isLoading: actions.isLoading,
      webpage,
    };
    return (
      <div className="add-webpage-section">
        <div className="add-webpage-headline">
          <nav
            className="pagination is-left"
            role="navigation"
            aria-label="pagination">
            {/* <Link to="/" className="pagination-next">Preview Your Webpage</Link> */}
            <ul className="pagination-list">
              <li>
                <NavLink
                  onClick={this.handleNavlinkClick}
                  to={`/dashboard/${username}/webpages/new/basic-info`}
                  activeClassName="is-current"
                  className="pagination-link"
                  aria-label="Basic Info">
                  <i className="fa fa-globe" aria-hidden="true" />
                    Basic Info
                </NavLink>
              </li>
              <li>
                <NavLink
                  onClick={this.handleNavlinkClick}
                  to={`/dashboard/${username}/webpages/new/contact-info`}
                  activeClassName="is-current"
                  className="pagination-link"
                  aria-label="Goto Contact Info">
                  <i className="fa fa-address-card-o" aria-hidden="true" />
                    Contact Info
                </NavLink>
              </li>
              <li>
                <NavLink
                  onClick={this.handleNavlinkClick}
                  to={`/dashboard/${username}/webpages/new/gallery`}
                  className="pagination-link"
                  activeClassName="is-current"
                  aria-label="Goto Gallery"
                  aria-current="page" >
                  <i className="fa fa-file-image-o" aria-hidden="true" />
                    Gallery
                </NavLink>
              </li>
              <li>
                <NavLink
                  onClick={this.handleNavlinkClick}
                  to={`/dashboard/${username}/webpages/new/offerings`}
                  className="pagination-link"
                  activeClassName="is-current"
                  aria-label="Goto Offerings">
                  <i className="fa fa-money" aria-hidden="true" />
                  Products/Services (Offerings)
                </NavLink>
              </li>
            </ul>
          </nav>
        </div>
        <div className="add-webpage-content">
          <Router>
            <Fragment>
              <Route
                exact
                path="/dashboard/:username/webpages/new/basic-info"
                render={() => (<BasicInfo
                  onSubmit={this.onSubmit}
                  user={user}
                  validationErrors={validationErrors}
                  handleErrorReset={this.handleErrorReset} />)}
              />
              <Route
                exact
                path="/dashboard/:username/webpages/new/contact-info"
                render={() => (<ContactInfo
                  onSubmit={this.submitContactInfo}
                  user={user}
                  validationErrors={validationErrors}
                  handleErrorReset={this.handleErrorReset} />)}
              />
              <Route
                exact
                path="/dashboard/:username/webpages/new/gallery"
                render={() => (<ImageUploader
                  onSubmit={this.submitImages}
                  user={user}
                  validationErrors={validationErrors}
                  handleErrorReset={this.handleErrorReset}
                  {...others}
                />)}
              />
              <Route
                exact
                path="/dashboard/:username/webpages/new/offerings"
                render={() => (<Offerings
                  onSubmit={this.submitOfferings}
                  user={user}
                  offerings={offerings}
                  validationErrors={validationErrors}
                  handleErrorReset={this.handleErrorReset}
                  handleOfferingImageSelection={
                    this.handleOfferingImageSelection}
                  offeringsFormRef={this.offeringsFormRef}
                  selectedImage={selectedImage}
                  handleSaveAndPreview={this.handleSaveAndPreview}
                />)}
              />
            </Fragment>
          </Router>
        </div>
      </div>
    );
  }
}

CreateWebpage.propTypes = {
  links: PropTypes.object,
  actions: PropTypes.object,
  user: PropTypes.object,
  history: PropTypes.object,
  webpage: PropTypes.object,
  username: PropTypes.string,
};

const mapStateToProps = ({ auth: { currentUser }, webpage }) => ({
  user: currentUser.user,
  webpage: webpage.newWebpage,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      createWebpage,
      updateWebpage,
      createWebpageOffering,
      isLoading,
      isComplete,
    },
    dispatch,
  ),
});

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(CreateWebpage));
