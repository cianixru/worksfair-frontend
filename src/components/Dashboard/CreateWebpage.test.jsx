import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup, fireEvent } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import CreateWebpage from './CreateWebpage';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({
  auth: { currentUser: {} },
  webpage: { newWebpage: {} }
});

describe('CreateWebpage.jsx', () => {
  // const validationErrors = {
  //   title: [
  //     'Title must be present',
  //   ]
  // };

  test('Should display the CreateWebpage Component', () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <CreateWebpage />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  test('should submit the create webpage form data', () => {
    const onSubmit = jest.fn();
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <Router>
          <CreateWebpage />
        </Router>
      </Provider>
    );

    const createBtn = getByTestId('create-webpage');
    createBtn.onclick = onSubmit;
    fireEvent.click(createBtn);

    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});
