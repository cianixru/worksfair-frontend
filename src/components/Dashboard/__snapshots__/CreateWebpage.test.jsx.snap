// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWebpage.jsx Should display the CreateWebpage Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="add-webpage-section"
      >
        <div
          class="add-webpage-headline"
        >
          <h3
            class="add-webpage-header"
          >
            <i
              aria-hidden="true"
              class="fa fa-globe"
            />
            Basic Info
          </h3>
        </div>
        <div
          class="add-webpage-content"
        >
          <h5>
            Website Owner
          </h5>
          <div
            class="media"
          >
            <div
              class="media-left"
            >
              <figure
                class="image is-32x32"
              >
                <img
                  alt="avatar"
                  class="is-rounded"
                  src="worksfair-avatar.png"
                />
              </figure>
            </div>
            <div
              class="media-content"
            />
          </div>
          <div
            class="add-webpage-form"
          >
            <form>
              <div
                class="field"
              >
                <h5>
                  Business Name
                  <i
                    class="tip"
                  >
                    <div
                      class="tip-content"
                    >
                      Keep it short. It is also your Webpage title.
                    </div>
                  </i>
                </h5>
                <div
                  class="control"
                >
                  <input
                    class="input is-medium"
                    data-testid="business-name"
                    name="title"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="field"
              >
                <h5>
                  Describe your business
                  <i
                    class="tip"
                  >
                    <div
                      class="tip-content"
                    >
                      Your customers will see this message on webpage.
                    </div>
                  </i>
                </h5>
                <div
                  class="control"
                >
                  <input
                    class="input is-medium"
                    data-testid="description"
                    name="description"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="field"
              >
                <h5>
                  Keywords
                  <i
                    class="tip"
                  >
                    <div
                      class="tip-content"
                    >
                      Some search keywords relating to your business.It will help people to find your business.
                    </div>
                  </i>
                </h5>
                <div
                  class="control"
                >
                  <input
                    class="input is-medium"
                    data-testid="business-keywords"
                    name="keywords"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="field"
              >
                <h5>
                  Colour
                  <i
                    class="tip"
                  >
                    <div
                      class="tip-content"
                    >
                      The colour theme for your webpage. the default is plain gray-scale
                    </div>
                  </i>
                </h5>
                <div
                  class="field"
                >
                  <div
                    class="control"
                  >
                    <div
                      class="select is-danger"
                    >
                      <select
                        name="colour"
                      >
                        <option>
                          Select Colour Theme
                        </option>
                        <option
                          value="is-danger"
                        >
                          ‚ù§Ô∏è Red
                        </option>
                        <option
                          value="is-primary"
                        >
                          üíö Green
                        </option>
                        <option
                          value="is-info"
                        >
                          üíô Blue
                        </option>
                        <option
                          value="is-warning"
                        >
                          üß° Orange
                        </option>
                        <option
                          value="is-dark"
                        >
                          üñ§ Black
                        </option>
                      </select>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="control"
              >
                <button
                  class="button is-danger is-medium is-rounded"
                  data-testid="create-webpage"
                  disabled=""
                  type="submit"
                >
                  Save
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="add-webpage-section"
    >
      <div
        class="add-webpage-headline"
      >
        <h3
          class="add-webpage-header"
        >
          <i
            aria-hidden="true"
            class="fa fa-globe"
          />
          Basic Info
        </h3>
      </div>
      <div
        class="add-webpage-content"
      >
        <h5>
          Website Owner
        </h5>
        <div
          class="media"
        >
          <div
            class="media-left"
          >
            <figure
              class="image is-32x32"
            >
              <img
                alt="avatar"
                class="is-rounded"
                src="worksfair-avatar.png"
              />
            </figure>
          </div>
          <div
            class="media-content"
          />
        </div>
        <div
          class="add-webpage-form"
        >
          <form>
            <div
              class="field"
            >
              <h5>
                Business Name
                <i
                  class="tip"
                >
                  <div
                    class="tip-content"
                  >
                    Keep it short. It is also your Webpage title.
                  </div>
                </i>
              </h5>
              <div
                class="control"
              >
                <input
                  class="input is-medium"
                  data-testid="business-name"
                  name="title"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <h5>
                Describe your business
                <i
                  class="tip"
                >
                  <div
                    class="tip-content"
                  >
                    Your customers will see this message on webpage.
                  </div>
                </i>
              </h5>
              <div
                class="control"
              >
                <input
                  class="input is-medium"
                  data-testid="description"
                  name="description"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <h5>
                Keywords
                <i
                  class="tip"
                >
                  <div
                    class="tip-content"
                  >
                    Some search keywords relating to your business.It will help people to find your business.
                  </div>
                </i>
              </h5>
              <div
                class="control"
              >
                <input
                  class="input is-medium"
                  data-testid="business-keywords"
                  name="keywords"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <h5>
                Colour
                <i
                  class="tip"
                >
                  <div
                    class="tip-content"
                  >
                    The colour theme for your webpage. the default is plain gray-scale
                  </div>
                </i>
              </h5>
              <div
                class="field"
              >
                <div
                  class="control"
                >
                  <div
                    class="select is-danger"
                  >
                    <select
                      name="colour"
                    >
                      <option>
                        Select Colour Theme
                      </option>
                      <option
                        value="is-danger"
                      >
                        ‚ù§Ô∏è Red
                      </option>
                      <option
                        value="is-primary"
                      >
                        üíö Green
                      </option>
                      <option
                        value="is-info"
                      >
                        üíô Blue
                      </option>
                      <option
                        value="is-warning"
                      >
                        üß° Orange
                      </option>
                      <option
                        value="is-dark"
                      >
                        üñ§ Black
                      </option>
                    </select>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="control"
            >
              <button
                class="button is-danger is-medium is-rounded"
                data-testid="create-webpage"
                disabled=""
                type="submit"
              >
                Save
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
